### DISCUSSION TITLE
Recommended method for outputing log_probs or top_logprobs

### DISCUSSION THREAD

**Discussion by de-abreu** [outlines-dev/outlines#1681] [Q&A]
Hello there,

I'm looking for a method to use Outlines with Mistral in a text classification task and output the probabilities of each label being picked. [Here](https://colab.research.google.com/drive/11_RKtnvI5NMc-1qiS1JSGAEc8G3NwKIQ?usp=sharing) is my sample experiment where I already implemented Outlines to the text classification task. How could I modify my experiment to implement log_probs?

Thanks!

**Comment by rlouf** [comment]
The recommended way is to implement a custom logits processor, a simplified version of https://github.com/dottxt-ai/outlines/pull/1408 and pass it when calling the model, see https://dottxt-ai.github.io/outlines/main/features/advanced/logits_processors/. We would definitely welcome a PR.
